src/main/java/budget/Main.java

```diff
@@ -0,0 +1,118 @@
+                    while (true) {
+                        try {
+                            income = Double.parseDouble(scanner.nextLine());
+                            break;
+                        } catch (NumberFormatException e) {
+                            System.out.print("Invalid input. Enter the total income as a number: ");
+                        }
+                    }
+                    System.out.print("Enter the start date (YYYY-MM-DD): ");
+                    String startDate = scanner.nextLine();
+                    System.out.print("Enter the end date (YYYY-MM-DD): ");
+                    String endDate = scanner.nextLine();
+                    manager.createBudget(name, income, startDate, endDate);
+                    break;
+                case 2:
+                    System.out.println("\nAdd an Expense:");
+                    System.out.print("Enter the name of the budget to add an expense: ");
+                    String budgetName = scanner.nextLine();
+                    System.out.print("Enter the expense amount: ");
+                    double amount = 0;
+                    while (true) {
+                        try {
+                            amount = Double.parseDouble(scanner.nextLine());
+                            break;
+                        } catch (NumberFormatException e) {
+                            System.out.print("Invalid input. Enter the amount as a number: ");
+                        }
+                    }
+                    System.out.print("Enter the expense description: ");
+                    String description = scanner.nextLine();
+                    System.out.print("Enter the date of the expense (YYYY-MM-DD): ");
+                    String date = scanner.nextLine();
+                    manager.addExpense(budgetName, amount, description, date);
+                    break;
+                case 3:
+                    System.out.println("\nView Saved Budgets:");
+                    manager.viewBudgets();
+                    break;
+                case 4:
+                    System.out.println("\nEdit a Budget:");
+                    System.out.print("Enter the name of the budget to edit: ");
+                    String editBudgetName = scanner.nextLine();
+                    System.out.print("What do you want to edit? (income/start_date/end_date): ");
+                    String field = scanner.nextLine();
+                    System.out.print("Enter the new value: ");
+                    String newValue = scanner.nextLine();
+                    manager.editBudget(editBudgetName, field, newValue);
+                    break;
+                case 5:
+                    System.out.println("\nEdit an Expense:");
+                    System.out.print("Enter the name of the budget to edit: ");
+                    String editExpenseBudgetName = scanner.nextLine();
+                    System.out.print("Enter the expense ID to edit: ");
```

Comment:
There is no way the user can know the expense IDs of their expenses? They are never shown the IDs and moreover, it's just bad UX in general to expect users to know IDs of objects in the application. Again, they should be shown a list of expenses and then asked to specify which they want to edit.

Reply:
Also, I see that there is no option to delete an expense. Important feature. Perhaps @EnoRich88 can take that up.